src/components/vanilla/charts/PivotTable/components/TableHead.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ReactElement } from 'react';
+import cn from 'classnames';
+import { TableHeaderType } from '../enums/TableHeaderType';
+
+export type Column = {
+  label: string;
+  key: string;
+  type: TableHeaderType;
+  children?: Column[];
+  minWidth?: number;
+}
+
+type Props = {
+  columns: Column[] | Array<Column[]>;
+}
+
+export default function TableHead({ columns }: Props) {
+
+  const getTotalColumns = (columns: Column[] | Array<Column[]>): number => {
+    return columns.reduce((acc, column) => {
+      if (Array.isArray(column)) {
+        return acc + getTotalColumns(column);
+      } else {
+        return acc + (column.children?.length || 1);
+      }
+    }, 0);
+  }
```

Comment:
Lets avoid declaring 'pure'  functions inside the components. Otherwise, every time the component re-renders, a new instance of that function will be created.

Having it outside is also more readable, easier to declare and also easier to test 🤓